Add authentication documentation page (#25)
tuwien-csd/damap-website: docs/manual/configuration/migration.md

---
title: Upgrading to v5.0.0
---

# Migration Guide: Upgrading to DAMAP v5.0.0

> DAMAP v5.0 changes how the system handles user logins and project permissions. This guide explains how to update your configuration files and how the new permission system affects your users.

Reviewer: Whole document needs `5.0` be changed to `5.0.0`

## 1. Configuration Changes

Older versions of DAMAP made hardcoded assumptions about how your login server worked. Version 5.0 gives you full control, but this means you must explicitly map your configuration settings to the new format.

> Every attribute related to user identity must now be clearly defined in your `application.yml` file. Please update your configuration by replacing the old attributes with the new ones, and adding the required new fields:

Reviewer: Needs to be rewritten with environment variables in mind and be made clearer.

| Old Attribute (v4.x)            | New Attribute (v5.0)               | What it means                                                                                                                               |
| :------------------------------ | :--------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------ |

> | `damap.auth.backend.url`        | `quarkus.oidc.auth-server-url`     | The address the backend uses to verify user logins.                                                                                         |

Reviewer: Please change new attribute to env variables.

| _(None - previously hardcoded)_ | `damap.auth.affiliations-claim`    | You must now explicitly state which field holds the user's institutional affiliation (e.g., whether they are a researcher at University A). |
| _(None - previously hardcoded)_ | `damap.auth.user-roles-claim-path` | The location in the login data where DAMAP can find the user's system roles (like admin rights).                                            |
| _(None - previously hardcoded)_ | `damap.auth.admin-role-name`       | The exact name of the role that gives a user full administrator privileges over the DAMAP instance.                                         |
| _(None - previously hardcoded)_ | `damap.auth.scope`                 | A list of permissions DAMAP asks for during login so it can read the user's basic profile information.                                      |

## 2. The New Permission System: Access vs. Contributors

The most noticeable change for everyday users in v5.0 is how editing permissions are granted.

### Why did this change?

> In the past, if a user added a colleague's name to the DMP's "Contributors" list, the system automatically gave that colleague permission to edit the document.

Reviewer: This is just not true and you should know that.

> However, modern login systems prioritize privacy by using hidden, randomized IDs for user accounts rather than public employee numbers. Because DAMAP can no longer safely guarantee that a public name on a document matches a specific, private login account, the system no longer grants automatic access based on the Contributors list.

Reviewer: "modern login systems", you mean the federation that still uses outdated SAML 😅
 "randomized IDs for user accounts rather than public employee numbers" TU Wien and Graz still give this information for example. 
"that a public name on a document matches" what name we are talking about ids? 

Sorry but that whole parapgraph needs to be rewritten (by hand).

The reason we needed to rework access management was that before we relied on the Idp giving us an internal university ID as a user ID, which we could match to contributors in the universities CRIS system - this linked contributors and identities. With the new cloud based system and PURE support, we cannot guarante this anymore since either new Unis wont or cannot provide us this ID or they use Pure, which uses internal IDs that are seperate from the university ID.

- **Contributors** are now just metadata. They are the names of people involved in the research project, which will be printed on the final DMP document.

> - **Access** is the actual, verifiable permission to open and edit the DMP in the software.

Reviewer: In theory this is correct, but the same was also true before 5.0.0 Needs to be rewritten to better get get your point across.

### How it works now

> Because adding a contributor no longer grants access, DMP owners must now manually invite their colleagues to collaborate.

Reviewer: See above

Here is the new workflow for granting access:

> 1. **The Colleague Logs In:** Before a colleague can be invited to edit a DMP, they must log into DAMAP at least once. This allows the system to securely register their name and email.

Reviewer: name email and **ID** which is the actual important part. I would just rewrite to:

```suggestion
1. **The Colleague Logs In:** Before a colleague can be invited to edit a DMP, they must log into DAMAP at least once. This allows the system to securely register them as users.
```

2. **The Owner Searches:** The DMP owner goes to the **Access** tab of their project and uses the search bar to find their colleague by name or email.

> 3. **The Owner Grants Access:** The owner selects the colleague and assigns them the "Editor" or "Viewer" role.

Reviewer: There is only Editor and Owner or no role (which you should know since you coded the migration)

!!! warning "Important Note for Existing DMPs"
Because of this security upgrade, people who previously relied on being a "Contributor" to edit a DMP will lose their editing access when you upgrade to v5.0.

> The original creator (Owner) of the DMP will need to go into the **Access** tab and manually search for and add their colleagues as Editors.

Reviewer: Are you sure - why would they lose access?

Author: In case the university only provides a pairwise ID but a university ID (such as TISS ID) was used before.
